File: DisplaBackend/DisplaBackend/Models/VentaVirtual.cs
﻿using System;
using System.Collections.Generic;

namespace DisplaBackend.Models
{
    public partial class VentaVirtual
    {
        public VentaVirtual()
        {
            VentaVirtualMovimientos = new HashSet<VentaVirtualMovimientos>();
        }

        public int Id { get; set; }
        public string Descripcion { get; set; }
        public decimal CantidadVendida { get; set; }
        public decimal CantidadEntregada { get; set; }
        public int? IdComprobante { get; set; }
        public int? IdRemito { get; set; }
        public int? IdArticulo { get; set; }
        public int? IdLente { get; set; }
        public bool Impreso { get; set; }
        public int? IdUsuario { get; set; }
        public int? IdServicio { get; set; }
        public decimal Monto { get; set; }

        public virtual ArticuloVario IdArticuloNavigation { get; set; }
        public virtual ComprobanteCliente IdComprobanteNavigation { get; set; }
        public virtual Lente IdLenteNavigation { get; set; }
        public virtual Remito IdRemitoNavigation { get; set; }
        public virtual Servicio IdServicioNavigation { get; set; }
        public virtual AspNetUsers IdUsuarioNavigation { get; set; }
        public virtual ICollection<VentaVirtualMovimientos> VentaVirtualMovimientos { get; set; }
    }
}

File: DisplaFrontend/src/app/model/ventaVirtual.ts
import { ArticuloVario } from './articuloVario';
import { ComprobanteCliente } from './comprobanteCliente';
import { Lente } from './lente';
import { Servicio } from './servicio';
import { Usuario } from './usuario';

export interface VentaVirtual {
    Id: number,
    Descripcion: string,
    CantidadVendida: number,
    CantidadEntregada: number,
    IdComprobante: number,
    IdArticulo: number,
    IdLente: number,
    IdServicio: number,
    Impreso: boolean, 
    IdUsuario: number,
    Monto: number,

    IdComprobanteNavigation: ComprobanteCliente,
    IdArticuloNavigation: ArticuloVario,
    IdLenteNavigation: Lente,
    IdServicioNavigation: Servicio,
    IdUsuarioNavigation: Usuario
}

